client/hotrod-client/src/main/java/org/infinispan/client/hotrod/configuration/Configuration.java

```diff
@@ -54,14 +57,16 @@
    private final StatisticsConfiguration statistics;
    private final TransactionConfiguration transaction;
    private final Features features;
+   private final Collection<SerializationContextInitializer> contextInitializers;
```

Comment:
```Collection``` is a bit too unspecific. It does not stress the fact that the initializers need to be executed in a predetermined order due to possible dependencies. We do not know anything about the dependecies but we assume the user knows, and provides the initializers to be executed in the right order. I'd suggest ```List```. 

Too bad I did not realize this when we did the core. I see now SerializationConfiguration in core need to have List also.

Reply:
That's a fair point, I'll change in the client to be List and I'll add another commit to update core